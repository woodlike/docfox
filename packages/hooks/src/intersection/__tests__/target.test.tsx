import * as React from 'react';
import { render } from '@testing-library/react';
import * as Intersection from '..';
import { TargetRenderData } from '..';

describe('<Intersection.Target />', () => {
  beforeAll(() => {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    (window as any).IntersectionObserver = jest.fn(() => ({
      observe: jest.fn(),
      unobserve: jest.fn(),
    }));
  });

  it('should have rendered with the initial TargetRenderData', () => {
    const renderFn = jest.fn().mockReturnValue(null);
    const { unmount } = render(
      <Intersection.Target>{renderFn}</Intersection.Target>,
    );
    const data: TargetRenderData = renderFn.mock.calls[0][0];
    expect(renderFn).toHaveBeenCalled();
    expect(data).toMatchSnapshot();
    unmount();
  });
});
