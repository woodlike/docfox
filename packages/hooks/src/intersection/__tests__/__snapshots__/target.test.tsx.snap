// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Intersection.Target /> should have rendered with the initial TargetRenderData 1`] = `
Object {
  "entry": Object {
    "boundingClientRect": Object {
      "bottom": 0,
      "height": 0,
      "left": 0,
      "right": 0,
      "top": 0,
      "width": 0,
      "x": 0,
      "y": 0,
    },
    "intersectionRatio": 0,
    "isIntersecting": false,
    "rootBounds": Object {
      "bottom": 0,
      "height": 0,
      "left": 0,
      "right": 0,
      "top": 0,
      "width": 0,
      "x": 0,
      "y": 0,
    },
    "time": 0,
  },
  "targetRef": Object {
    "current": null,
  },
}
`;
